Game mechanics

droid-101 opened the pull request.

droid-101 committed:
Added display()

Added a display method which prints out the TicTacToe grid with cells in which Xs and Os will be placed.

droid-101 committed:
Added winCheck()

Added winCheck() method to test the conditions of the game to check if a player has won.

droid-101 committed:
Fixed Bugs

Fixed the issue with turn skipping and the issue with cell overwriting by different players.

bricktrooper commented on TicTacToe.java:
```diff
@@ -0,0 +1,229 @@
+        cellLetters[5] = '6';
+        cellLetters[6] = '7';
+        cellLetters[7] = '8';
+        cellLetters[8] = '9';
```
Why not use a for loop here?

bricktrooper commented on TicTacToe.java:
```diff
@@ -0,0 +1,229 @@
+            {
+                playerOneMove = playerOne.nextMove();
+
+                if (openCheck(cellLetters[playerOneMove - 1]) == true)
```
Quick tip: you don't need == true here.  Since openChek will already return true or false

bricktrooper commented on TicTacToe.java:
```diff
@@ -0,0 +1,229 @@
+                }
+                else
+                {
+                    System.out.println("The selected cell is taken. Please enter a different cell.");
```
Why no immediately ask player 1 to enter a new value instead of going through the whole loop?  Then you don't need the playerOneTurn and playerTwoTurn variables.

bricktrooper commented on TicTacToe.java:
```diff
@@ -0,0 +1,229 @@
+
+            if (gameEnd == false)
+            {
+                if (playerTwoTurn == true)
```
If you want to check ==false, you can do !playerTwoTurn, which reads "not player 2's turn" which makes more sense

bricktrooper replied:
fix == true

bricktrooper commented on TicTacToe.java:
```diff
@@ -0,0 +1,229 @@
+
+    public static boolean openCheck(char cellContent)
+    {
+
```
Get rid of this random space

droid-101 committed:
Added Makefile and fixed bugs

-Makefile was created.
-Program now asks players for their names.
-The array creator was put into a for loop.
